profiles/tests/test_profiles_graphql_api.py

```diff
@@ -3358,3 +3365,118 @@ def test_user_cannot_claim_claimable_profile_with_existing_profile(rf, user_gql_
 
     assert "errors" in executed
     assert executed["errors"][0]["extensions"]["code"] == API_NOT_IMPLEMENTED_ERROR
+
+
+class TestTemporaryProfileReadAccessTokenCreation:
+    query = """
+        mutation {
+            createMyProfileTemporaryReadAccessToken(input: { }) {
+                temporaryReadAccessToken {
+                    token
+                    expiresAt
+                }
+            }
+        }
+    """
+
+    def test_normal_user_can_create_temporary_read_access_token_for_profile(
+        self, rf, user_gql_client
+    ):
+        ProfileFactory(user=user_gql_client.user)
+        request = rf.post("/graphql")
+        request.user = user_gql_client.user
+
+        executed = user_gql_client.execute(self.query, context=request)
+
+        token_data = executed["data"]["createMyProfileTemporaryReadAccessToken"][
+            "temporaryReadAccessToken"
+        ]
+
+        # Check that an UUID can be parsed from the token
+        uuid.UUID(token_data["token"])
+
+        actual_expiration_time = datetime.fromisoformat(token_data["expiresAt"])
+        approximate_expected_expiration_time = timezone.now() + timedelta(days=2)
+        expiration_time_delta = (
+            approximate_expected_expiration_time - actual_expiration_time
+        )
+        assert timedelta(0) <= expiration_time_delta < timedelta(seconds=1)
```

Comment:
This might fail if you do not freeze the time.
It would be easier to read if it would be encapsulated in a function

asset isTokenExpirationTimeLessThan(int delta_in_milisecond)

Reply:
I now have code like this (not yet pushed):

```python
actual_expiration_time = datetime.fromisoformat(token_data["expiresAt"])
expected_expiration_time = timezone.now() + timedelta(days=2)
assert_almost_equal(
    actual_expiration_time, expected_expiration_time, timedelta(seconds=1)
)
```

Do you consider this more readable?